addons/subscription/models/subscription.py

```diff
@@ -0,0 +1,123 @@
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+def _get_document_types(self):
+    self.env.cr.execute('select m.model, s.name from subscription_document s, ir_model m WHERE s.model_id = m.id order by s.name')
+    return self.env.cr.fetchall()
+
+class SubscriptionDocument(models.Model):
+    _name = "subscription.document"
+    _description = "Subscription Document"
+
+    name = fields.Char(required=True)
+    active = fields.Boolean(help="If the active field is set to False, it will allow you to hide the subscription document without removing it.", default=True)
+    model_id = fields.Many2one('ir.model', string="Object", required=True, oldname='model')
+    field_ids = fields.One2many('subscription.document.fields', 'document_id', string='Fields', copy=True)
+
+class SubscriptionDocumentFields(models.Model):
+    _name = "subscription.document.fields"
+    _description = "Subscription Document Fields"
+    _rec_name = 'field_id'
+
+    field_id = fields.Many2one('ir.model.fields', domain="[('model_id', '=', parent.model_id)]", required=True, oldname='field')
+    value = fields.Selection([('false', 'False'), ('date', 'Current Date')], string='Default Value', help="Default value is considered for field when new document is generated.")
+    document_id = fields.Many2one('subscription.document', string='Subscription Document', ondelete='cascade')
+
+
+class Subscription(models.Model):
+    _name = "subscription.subscription"
+    _description = "Subscription"
+
+    name = fields.Char(required=True)
+    active = fields.Boolean(help="If the active field is set to False, it will allow you to hide the subscription without removing it.", default=True)
+    partner_id = fields.Many2one('res.partner', string='Partner')
+    notes = fields.Text(string='Internal Notes')
+    user_id = fields.Many2one('res.users', string='User', required=True, default=lambda self: self.env.user)
+    interval_number = fields.Integer(string='Internal Qty', default=1)
+    interval_type = fields.Selection([('days', 'Days'), ('weeks', 'Weeks'), ('months', 'Months')], string='Interval Unit', default='months')
+    exec_init = fields.Integer(string='Number of Documents')
+    date_init = fields.Datetime(string='First Date', default=fields.Datetime.now)
+    state = fields.Selection([('draft', 'Draft'), ('running', 'Running'), ('done', 'Done')], string='Status', copy=False, default='draft')
+    doc_source = fields.Reference(string='Source Document', required=True, selection=_get_document_types, help="User can choose the source document on which he wants to create documents", default=False)
+    doc_line_ids = fields.One2many('subscription.subscription.history', 'subscription_id', string='Documents created', readonly=True, oldname='doc_lines')
+    cron_id = fields.Many2one('ir.cron', string='Cron Job', help="Scheduler which runs on subscription", states={'running':[('readonly',True)], 'done':[('readonly', True)]})
```

Comment:
Missing white-space at many places after colons/commas. Need to check the whole file.